### PR TITLE
[scoped-custom-element-registry] Don't try to use scopeForElement with insertAdjacentHTML

### LINKED ISSUE #444: [scoped-custom-element-registry]: insertAdjacentHTML throws exception
Exception is thrown here:
https://github.com/webcomponents/polyfills/blob/041aff1a2408a5854c880e45dfad150556e51bcb/packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js#L345

Reason: native `insertAdjacentHTML` returns `undefined`, but `WeakMap`s only support objects as keys:
- https://w3c.github.io/DOM-Parsing/#extensions-to-the-element-interface
- step 4 in https://tc39.es/ecma262/#sec-weakmap.prototype.set

Can be reproduced with the following example, which should spell out "success" (I can get the example to work by commenting out the line `installScopedCreationMethod(Element, 'insertAdjacentHTML');`):
```html
<span>u</span><span id="host"></span>
<script>
    class MyWidgetGlobal extends HTMLElement {
        constructor() {
            super()
            this.attachShadow({ mode: 'open' })
            this.shadowRoot.innerHTML = '<span>s</span>'
        }
    }
    class MyWidgetScoped extends HTMLElement {
        constructor() {
            super()
            this.attachShadow({ mode: 'open' })
            this.shadowRoot.innerHTML = '<span>c</span>'
        }
    }
    window.customElements.define('my-widget', MyWidgetGlobal)
    try {
        const span = document.querySelector('span')
        span.insertAdjacentHTML('beforebegin', '<my-widget></my-widget>')
        // next line won't execute
        span.insertAdjacentText('afterend', 'c')
    } catch (ex) {
        console.log(ex)
    }
    const scope = new CustomElementRegistry()
    scope.define('my-widget', MyWidgetScoped)
    const host = document.querySelector('#host')
    const shadowRoot = host.attachShadow({ mode: 'open', customElements: scope })
    shadowRoot.innerHTML = '<span>es</span>'
    try {
        const span = shadowRoot.querySelector('span')
        span.insertAdjacentHTML('beforebegin', '<my-widget></my-widget>')
        // next line won't execute
        span.insertAdjacentText('afterend', 's')
    } catch (ex) {
        console.log(ex)
    }
</script>
```

### PR DESCRIPTION
**Pull request by kevinpschaaf**
FIxes #444

### CODE REVIEW

File: packages/scoped-custom-element-registry/test/Element.test.html.js [hunk 1]
```diff
@@ -55,4 +78,30 @@ describe('Element', () => {
       });
     });
   });
+
+  describe('insertAdjacentHTML', () => {
+    it('should upgrade an element defined in the custom registry', () => {
+      const {tagName, CustomElementClass} = getTestElement();
+      const registry = new CustomElementRegistry();
+      const shadowRoot = getShadowRoot(registry);
+      const $el = getHTML('<div></div>', shadowRoot);
```
Comment (justinfagnani) [inline packages/scoped-custom-element-registry/test/Element.test.html.js:87]: why the `$` prefix?

File: packages/scoped-custom-element-registry/test/Element.test.html.js [hunk 2]
```diff
@@ -55,4 +78,30 @@ describe('Element', () => {
       });
     });
   });
+
+  describe('insertAdjacentHTML', () => {
+    it('should upgrade an element defined in the custom registry', () => {
+      const {tagName, CustomElementClass} = getTestElement();
+      const registry = new CustomElementRegistry();
+      const shadowRoot = getShadowRoot(registry);
```
Comment (justinfagnani) [inline packages/scoped-custom-element-registry/test/Element.test.html.js:86]: OT: Do `getShadowRoot()` and `getHTML()` coordinate via shared global state? Makes for non-obvious flow here...

**Review by justinfagnani — APPROVED** [review]

File: packages/scoped-custom-element-registry/test/Element.test.html.js (hunk 2 above)
Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/test/Element.test.html.js:86]: No, you pass the root you want to innerHTML into into `getHTML`, and it just returns out the first element.

File: packages/scoped-custom-element-registry/test/Element.test.html.js (hunk 1 above)
Comment (kevinpschaaf) [reply packages/scoped-custom-element-registry/test/Element.test.html.js:87]: Just copy/paste from the existing style of the tests. We can consider a separate task to normalize this repo the style used in the rest of the monorepo.

**Pull request merged**